File: python/services/github/github_types.py
# -*- coding: utf-8 -*-
#
#  2023 ozkary.com.
#
#  GitHub API data types
#

from dataclasses import dataclass

@dataclass
class Issue:
    id: str
    title: str
    body: str

@dataclass
class Parameter:
    label: str
    state: str

File: csharp/CodeGeneration/Services/Github/Github.Types.cs
namespace CodeGeneration.Services.GitHub {

    /// <summary>
    /// GitHub issue interface
    /// </summary>
    internal class Issue
    {
        public int id { get; set; }
        public int number { get; set; }
        public string? title { get; set; }
        public string? body { get; set; }
    }

    /// <summary>
    /// GitHub API parameter interface
    /// </summary>
    internal class Parameter {    
        public string Label {get; set;} = "";
        public string State {get; set;} = "";
    }

}